### ISSUE TITLE
Wrong save file size created

### ISSUE THREAD

**Issue by nuxa17** [DS-Homebrew/TWiLightMenu#2486]
A save file of an incorrect size is being created for Pic Pic, making it impossible to save.

**Flashcard or Exploits?** Ace3DS+ clone (X in 1 games), Ace3DS+ R4iLS 1.62
**Base Console:** Nintendo 3DS XL, 11.17.0-50E

**TWiLight Menu++ Version:** v27.11.0
**TWiLight Menu++ Theme:** Tested with Nintendo 3DS and DSi themes

### Expected behavior

Pic Pic has a save file of size 64kb. When using the original kernel without TWiLight Menu++, the created save file has a size of 512kb (probably a default size, see Notes). Both of them work correctly. When using TWiLight Menu++ with Autoboot and Flashcart Loader, the save file should be created with the same size.

### Actual behavior

When using TWiLight Menu++ with Autoboot and Flashcart Loader, the created save file has a size of 8kb.

### Steps to reproduce

1. On an empty SD card, move the files from the kernel zip file (link in notes).
2. Install TWiLight Menu++ with the Autoboot and Flashcart Loader files for Ace3DS+.
3. Move the game to the roms/nds folder.
4. Run the game. The file is created before loading, so it can be closed when the first credits appear.
5. The save file created will have 8kb. Also, it will be placed outside the `saves` folder.

### Tests

- Different SD card, completely empty. Following the same steps produces the same output.
- Tested with original AOS instead of R4iLS, saves fine.
- Originally tested with TWiLight Menu++ v27.5.0.

### Notes

- MelonDS uses a list with the specific save size per game, and creates a file of 64kb.
- This is not the first time I have problems with save files. I think the problem resides in the Flashcart Loader, since it uses a different kernel (see #1783). If there is a better place to submit this bug, please let me know.
- For testing: the game saves for the first time when a new user is created.
- R4iLS kernel taken from the [flashcart archive](https://archive.flashcarts.net/Ace3DS+_R4iLS).

**Comment by nuxa17** [comment]
More tests:
- Removing `_wfwd/savelist.bin` creates a 512kb save file.
- Copying `__rpg/savesize.bin` to `_wfwd` creates a 8kb save file.
- Replacing `_wfwd/savelist.bin` with `__rpg/savesize.bin` and renaming it creates a 512kb save file.

The format of `savesize.bin` and `savelist.bin` is different, but I wanted to try it out just in case. On the other hand, when using the original kernel, you can change the save size manually. Changes per game are stored in `__rpg/gamedata.bin` which uses the same format as `savelist.bin`.

**Comment by nuxa17** [comment]
Tested with two more games of the same save size according to `__wfwd/savelist.bin`:
- Meteos.  MelonDS: 8kb.
- Top Spin 3. MelonDS: 64kb.

These games generate a 8kb save file using the Flashcart Loader. 

After formatting the SD to a cluster size of 32kb, the generated save file for Pic Pic has a size of 512 bytes. The original kernel still creates a file of 512kb, and nds-bootstrap generates the save files with correct sizes.

**Comment by nuxa17** [comment]
The issue persist in v27.11.1

**Comment by RocketRobz** [comment]
What could be done is for TWLMenu++ to create the save file instead of the flashcard kernel, but then afaik, YSMenu saves it's per-game settings at the end of the `.sav` file, so it probably won't work properly for flashcards which use YSMenu as the flashcard kernel.

**Comment by nuxa17** [comment]
Judging by the variation of sizes depending of the card's format, I'm worried that there could be more issues with this particular loader. If that's the case, an option could be to detect if this particular loader is in use, and generate the saves if that's the case.

**Comment by Elias1878** [comment]
Have you found a solution to this save file problem? My Ace3DS+ cart is also doing this.

**Comment by nuxa17** [comment]
Nope, I didn't have much time to try other things. As far as I can remember, GodMode9i also gave me problems when writing data.
